packages/go_router_builder/example/lib/shell_route_example.dart

```diff
@@ -0,0 +1,153 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: public_member_api_docs
+
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+
+part 'shell_route_example.g.dart';
+
+void main() => runApp(App());
+
+class App extends StatelessWidget {
+  App({super.key});
+
+  @override
+  Widget build(BuildContext context) => MaterialApp.router(
+        routerConfig: _router,
+      );
+
+  final GoRouter _router = GoRouter(
+    routes: $appRoutes,
+    initialLocation: '/foo',
+  );
+}
+
+class HomeScreen extends StatelessWidget {
+  const HomeScreen({super.key});
+
+  @override
+  Widget build(BuildContext context) => Scaffold(
+        appBar: AppBar(title: const Text('foo')),
+      );
+}
+
+@TypedShellRoute<MyShellRouteData>(
+  routes: <TypedRoute<RouteData>>[
+    TypedGoRoute<FooRouteData>(path: '/foo'),
+    TypedGoRoute<BarRouteData>(path: '/bar'),
+  ],
+)
+class MyShellRouteData extends ShellRouteData {
```

Comment:
Can you also add navigatorKey and parentNavigatorKey example? it can be in this example or a separate one.

Reply:
Done in [Add example of navigation with keys](https://github.com/flutter/packages/pull/3439/commits/3013a05f73a178e8926b3a7f3f4e7158a373ac0b).

Here's a quick demo of the example. I use the navigator key to make sure that the dialog route is displayed over the entire screen instead of just the right of the navigator rail:
![nav with keys](https://user-images.githubusercontent.com/2680243/229681726-ce1b1c73-fd42-4447-9add-d72ecf1d4f26.gif)

I've also added a section in the README in [Add section about navigator keys in README](https://github.com/flutter/packages/pull/3439/commits/3376c7367804892dbe9cc1745ff2966778107f94)